src/Plugin.cpp

```diff
@@ -93,13 +78,68 @@ void Plugin::deInit() {
     mSolarSystem->unregisterAnchor(ring);
   }
 
-  for (auto const& ringNode : mRingNodes) {
-    mSceneGraph->GetRoot()->DisconnectChild(ringNode.get());
-  }
+  mAllSettings->onLoad().disconnect(mOnLoadConnection);
+  mAllSettings->onSave().disconnect(mOnSaveConnection);
 
   logger().info("Unloading done.");
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+void Plugin::onLoad() {
+  // Read settings from JSON.
+  from_json(mAllSettings->mPlugins.at("csp-rings"), mPluginSettings);
```

Comment:
I'm not sure if this is the correct code, but it looks cleaner with an assignment in my opinion.

```suggestion
  mPluginSettings = mAllSettings->mPlugins.at("csp-rings").get<Plugin::Settings>();
```

Reply:
This wouldn't work. By passing mPluginSettings as reference, all the onChange signals of the properties get called when they change due to the loading. If we would overwrite the mPluginSettings, all handlers would be disconnected instead.